src/applications/mhv-secure-messaging/tests/e2e/contact-list-tests/secure-messaging-contact-list-multi-facility-navigate-away-errors.cypress.spec.js

```diff
@@ -0,0 +1,102 @@
+import SecureMessagingSite from '../sm_site/SecureMessagingSite';
+import PatientInboxPage from '../pages/PatientInboxPage';
+import ContactListPage from '../pages/ContactListPage';
+import { AXE_CONTEXT, Paths } from '../utils/constants';
+import mockEhrData from '../fixtures/userResponse/vamc-ehr-cerner-mixed.json';
+import mockMixedCernerFacilitiesUser from '../fixtures/userResponse/user-cerner-mixed.json';
+import mockFacilities from '../fixtures/facilityResponse/cerner-facility-mock-data.json';
+import mockMixRecipients from '../fixtures/multi-facilities-recipients-response.json';
+
+describe('SM Single Facility Contact list', () => {
+  beforeEach(() => {
+    SecureMessagingSite.login(
+      mockEhrData,
+      true,
+      mockMixedCernerFacilitiesUser,
+      mockFacilities,
+    );
+    PatientInboxPage.loadInboxMessages();
+    ContactListPage.loadContactList(mockMixRecipients);
+    ContactListPage.selectAllCheckBox();
+  });
+
+  it('navigate away using secondary navigation', () => {
+    cy.injectAxe();
+    cy.axeCheck(AXE_CONTEXT);
+
+    cy.window().then(win => {
+      const beforeUnloadStub = cy.stub();
+
+      win.addEventListener('beforeunload', beforeUnloadStub);
+
+      cy.visit(`${Paths.UI_MAIN}/medications/about`);
+
+      cy.then(() => {
+        expect(beforeUnloadStub).to.have.been.called;
+        expect(win.location.href).to.eq(
+          `http://localhost:3001${Paths.UI_MAIN}/medications/about`,
```

Comment:
@fazilqa do we know of this validation of `localhost` wil not be a problem when tests are run in a deployment pipeline?

Reply:
https://dsva.slack.com/archives/C03T8JY2DM2/p1725894347896519?thread_ts=1725471203.443519&cid=C03T8JY2DM2

Reply:
as I can see right now those tests are passing in CI runs